service/server.py

```diff
@@ -117,6 +117,20 @@ def get_titles_for_address(address):
     return jsonify(result)
 
 
+@app.route('/save_search_request', methods=['POST'])
+def save_search_request():
+    # N.B.: "request.form" is a 'multidict', so need to flatten it first; assume single value per key.
+    form_dict = request.form.to_dict()
+    db_access.save_user_search_details(form_dict)
```

Comment:
do we need to safety check the contents of the dict before sending to db_access or get db_access to not trust us and do the checks there?